src/com/lftechnology/java_training/realworldproblem/VariousCarTester.java

```diff
@@ -0,0 +1,137 @@
+package com.lftechnology.java_training.realworldproblem;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+import java.util.logging.Logger;
+
+/**
+ * This class performs various feature and functionalities of {@link ElectricCar} and {@link PetrolCar}
+ * 
+ * @author Utsab Neupane<utsabneupane@lftechnology.com>
+ * 
+ */
+public class VariousCarTester {
+	private static Logger LOGGER = Logger.getLogger(ElectricCar.class.getName());
+	private static Scanner input = new Scanner(System.in);
+	private static int choiceInput = 0;
+
+	public static void main(String[] args) {
+		String inputChoice = "";
```

Comment:
Now let not handle logic in the client call(that is main method), let service class to decide which class to simulate.
Lets think a real time senario, When we go for bike servicing, it is not the bike owner who decides what type of services is to be done to the bike, it auto mobile service who decides this.